### ISSUE TITLE
Array indexes in phys_grid.F90

### ISSUE THREAD

**Issue by amametjanov** [E3SM-Project/E3SM#3852]
Appears to be a copy-paste error. Array indexes need to be declared as `integer`, not `real(r8)`.

Errors (with cray compiler):
```
ftn-319 crayftn: ERROR PROC_HEAP_ADJUST, File = ../../../gpfs/mira-home/azamatm/repos/E3SM/components/cam/src/physics/cam/phys_grid.F90, Line = 6519, Column = 19
  A subscript must be a scalar integer expression.


ftn-319 crayftn: ERROR PROC_HEAP_ADJUST, File = ../../../gpfs/mira-home/azamatm/repos/E3SM/components/cam/src/physics/cam/phys_grid.F90, Line = 6564, Column = 22
  A subscript must be a scalar integer expression.


ftn-319 crayftn: ERROR PROC_HEAP_ADJUST, File = ../../../gpfs/mira-home/azamatm/repos/E3SM/components/cam/src/physics/cam/phys_grid.F90, Line = 6602, Column = 25
  A subscript must be a scalar integer expression.


ftn-319 crayftn: ERROR PROC_HEAP_ADJUST, File = ../../../gpfs/mira-home/azamatm/repos/E3SM/components/cam/src/physics/cam/phys_grid.F90, Line = 6621, Column = 25
  A subscript must be a scalar integer expression.

Cray Fortran : Version 10.0.1 (20200515195441_eae63ace5824bc000ce707e461011e3af92af44b)
Cray Fortran : Compile time:  0.1608 seconds
Cray Fortran : 6644 source lines
Cray Fortran : 5 errors, 12 warnings, 0 other messages, 0 ansi
Cray Fortran : "explain ftn-message number" gives more information about each message.
gmake[2]: *** [cmake/atm/CMakeFiles/atm.dir/build.make:3965: cmake/atm/CMakeFiles/atm.dir/__/__/cam/src/physics/cam/phys_grid.F90.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
gmake[2]: Leaving directory '/lus/theta-fs0/projects/ClimateEnergy_4/azamatm/tests/20200925-cray-integr/ERP_Ld3.ne30_oECv3_ICG.A_WCYCL1850S.theta_cray.20200926_013819_45pfr0/bld/cmake-bld'
gmake[1]: *** [CMakeFiles/Makefile2:279: cmake/atm/CMakeFiles/atm.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
```

Patch:
```
diff --git a/components/cam/src/physics/cam/phys_grid.F90 b/components/cam/src/physics/cam/phys_grid.F90
index 880de2e819..96a6ec6da8 100644
--- a/components/cam/src/physics/cam/phys_grid.F90
+++ b/components/cam/src/physics/cam/phys_grid.F90
@@ -6457,9 +6457,9 @@ logical function phys_grid_initialized ()
    integer :: proc_last    ! process id of last element
    integer :: proc_updated ! process id of updated element
    integer :: proc_i       ! process id of current element
-   real(r8):: proc_ip      ! process id of parent of current element
-   real(r8):: proc_il      ! process id of left child of current element
-   real(r8):: proc_ir      ! process id of right child of current
+   integer :: proc_ip      ! process id of parent of current element
+   integer :: proc_il      ! process id of left child of current element
+   integer :: proc_ir      ! process id of right child of current
                            !  element

    integer :: last_nonleaf ! index of last non-leaf in heap
```

**Comment by worleyph** [comment]
That's embarrassing. Sorry about that. Intel compiler is too forgiving.

**Comment by worleyph** [comment]
@amametjanov , are you submitting a PR with this fix, or do you want me to?

**Comment by amametjanov** [comment]
Should be fixed when this commit https://github.com/E3SM-Project/E3SM/pull/3855/commits/e8fb7a87b75320a3dbf76023a9daaa07a9d3938e goes in.

**Comment by amametjanov** [comment]
Fixed by E3SM-Project/E3SM#3963
